libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackHelper.cs

```diff
@@ -0,0 +1,356 @@
+        public static async Task WriteAsync(HttpResponse response, HttpStatusCode code, string text, Encoding encoding, CancellationToken cancellationToken = default)
+        {
+            if (response == null)
+            {
+                throw new ArgumentNullException(nameof(response));
+            }
+
+            if (text == null)
+            {
+                throw new ArgumentNullException(nameof(text));
+            }
+
+            if (encoding == null)
+            {
+                throw new ArgumentNullException(nameof(encoding));
+            }
+
+            response.ContentType = "text/plain";
+            response.StatusCode = (int)code;
+
+            var data = encoding.GetBytes(text);
+
+            await response.Body.WriteAsync(data, 0, data.Length, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Creates an activity based on the slack event payload.
+        /// </summary>
+        /// <param name="slack">The payload of the slack event.</param>
+        /// <returns>An activity containing the event data.</returns>
+        public static Activity PayloadToActivity(SlackPayload slack)
+        {
+            if (slack == null)
+            {
+                return null;
+            }
+
+            var activity = new Activity()
+            {
+                Timestamp = default(DateTime),
+                ChannelId = "slack",
+                Conversation = new ConversationAccount()
+                {
+                    Id = slack.Channel.id,
+                },
+                From = new ChannelAccount()
+                {
+                    Id = slack.Message?.BotId ?? slack.User.id,
+                },
+                Recipient = new ChannelAccount()
+                {
+                    Id = null,
+                },
+                ChannelData = slack,
+                Text = null,
+                Type = ActivityTypes.Event,
+            };
+
+            if (slack.ThreadTs != null)
+            {
+                activity.Conversation.Properties["thread_ts"] = slack.ThreadTs;
+            }
+
+            if (slack.Actions != null && (slack.Type == "block_actions" || slack.Type == "interactive_message"))
+            {
+                activity.Type = ActivityTypes.Message;
+                activity.Text = slack.Actions[0].Value;
+            }
+
+            return activity;
+        }
+
+        /// <summary>
+        /// Creates an activity based on the slack event data.
+        /// </summary>
+        /// <param name="slack">The data of the slack event.</param>
+        /// <param name="client">The Slack client.</param>
+        /// <param name="cancellationToken">A cancellation token for the task.</param>
+        /// <returns>An activity containing the event data.</returns>
+        public static async Task<Activity> EventToActivityAsync(SlackEvent slack, SlackClientWrapper client, CancellationToken cancellationToken)
```

Comment:
Consider renaming slack param to slackEvent

Reply:
Done. Thanks!